azure/converters/rules_test.go

```diff
@@ -66,28 +68,28 @@ func TestSecurityRuleToSDK(t *testing.T) {
 		SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
 			Access: network.SecurityRuleAccessAllow,
 		},
-		Protocol: network.SecurityRuleProtocolAsterisk,
+		Protocol:  network.SecurityRuleProtocolAsterisk,
 		Direction: network.SecurityRuleDirectionInbound,
 	}
 
 	actualResult = SecurityRuleToSDK(rule)
-	assert.Equal(t, expectedResult, actualResult)
+	g.Expect(actualResult).To(gomega.Equal(expectedResult))
 
 	// Test case for a security rule with invalid protocol
 	rule = infrav1.SecurityRule{
-		Name: "test-rule",
-		Protocol: "invalid-protocol",
+		Name:      "test-rule",
+		Protocol:  "invalid-protocol",
 		Direction: infrav1.SecurityRuleDirectionInbound,
 	}
 
-	assert.PanicsWithError(t, "invalid protocol value", func() { SecurityRuleToSDK(rule) })
+	g.Expect(func() { SecurityRuleToSDK(rule) }).To(gomega.PanicWith("invalid protocol value"))
 
 	// Test case for a security rule with invalid direction
 	rule = infrav1.SecurityRule{
-		Name: "test-rule",
-		Protocol: infrav1.SecurityGroupProtocolAll,
+		Name:      "test-rule",
+		Protocol:  infrav1.SecurityGroupProtocolAll,
 		Direction: "invalid-direction",
 	}
 
-	assert.PanicsWithError(t, "invalid direction value", func() { SecurityRuleToSDK(rule) })
+	g.Expect(func() { SecurityRuleToSDK(rule) }).To(gomega.PanicWith("invalid direction value"))
```

Comment:
Instead of having repeated code here, you should refactor this to use a list of test cases, then iterate through the list and perform these checks.

Example: https://github.com/willie-yao/cluster-api-provider-azure/blob/azuremachine-controller-test/azure/converters/subnets_test.go#L28